events/renderers/docx.py

```diff
@@ -0,0 +1,269 @@
+from collections import OrderedDict
+
+import dateutil
+from bs4 import BeautifulSoup
+from django.utils.text import slugify
+from django.utils.timezone import localtime
+from docx import Document
+from rest_framework import renderers
+from rest_framework.exceptions import APIException
+
+
+def get_any_language(dictionary, default='fi', language_codes=None):
+    if not language_codes:
+        # Default order for when a language isn't found.
+        language_codes = list([default, 'fi', 'sv', 'en'])
+
+    for language_code in language_codes:
+        content = dictionary.get(language_code)
+        if content:
+            return content
+    return ''
+
+
+def get_description(event):
+    if event['short_description']:
+        description_with_html = event['short_description']
+    elif event['description']:
+        description_with_html = event['description']
+    return description_with_html
+
+
+def remove_html(html):
+    # Fake tags added here so we ensure that we get all the text. Using
+    # only a string here apparently results in nothing being returned.
+    return BeautifulSoup(
+        '<fake>%s</fake>' % html,
+        'xml',
+    ).text
+
+
+def get_price(raw_event):
+    try:
+        return dict(raw_event['offers'][0])['price']
+    except:
+        return None
+
+
+class EventParser():
+    def __init__(self):
+        self.earliest_date = None
+        self.latest_date = None
+
+    def parse_event(self, raw_event):
+        '''
+        Turns a raw event into a dictionary with only needed fields.
+        Keeps track of earliest and latest times encountered.
+        '''
+        start_time = localtime(raw_event['start_time_obj'])
+        end_time = localtime(raw_event['end_time_obj'])
+
+        if self.earliest_date is None or start_time.date() < self.earliest_date:
+            self.earliest_date = start_time.date()
+
+        if self.latest_date is None or end_time.date() > self.latest_date:
+            self.latest_date = start_time.date()
+
+        price = get_price(raw_event)
+        if price is not None:
+            price = get_any_language(price)
+        else:
+            price = ''
+
+        return {
+            'name': get_any_language(raw_event['name']),
+            'description': remove_html(
+                get_any_language(get_description(raw_event))
+            ),
+            'location': raw_event['location'],
+            'start_time': start_time,
+            'end_time': end_time,
+            'price': price,
+        }
+
+
+def group_by_location(events):
+    # We want to put events in the same location under the same headline,
+    # so we create a dictionary where the places are the keys and the events
+    # are the values.
+    locations = {}
+    for event in events:
+        place = event['location']
+        if place in locations:
+            locations[place].append(event)
+        else:
+            locations[place] = [event]
+    return locations
+
+
+class DateRange():
```

Comment:
What would be a good place for it? The use case is so specific to the docx renderer that I don't think it has much reuse value unless you want to print a similar PDF file or something.